src/EditorFeatures/Core/Implementation/ExtractMethod/AbstractExtractMethodCommandHandler.cs

```diff
@@ -72,28 +72,27 @@ public VSCommanding.CommandState GetCommandState(ExtractMethodCommandArgs args)
 
         public bool ExecuteCommand(ExtractMethodCommandArgs args, CommandExecutionContext context)
         {
-            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Applying_Extract_Method_refactoring))
+            var document = args.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document == null)
             {
-                var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChangesAsync(
-                    context.OperationContext).WaitAndGetResult(context.OperationContext.UserCancellationToken);
-                if (document == null)
-                {
-                    return false;
-                }
+                return false;
+            }
 
-                var supportsFeatureService = document.Project.Solution.Workspace.Services.GetService<IDocumentSupportsFeatureService>();
-                if (!supportsFeatureService.SupportsRefactorings(document))
-                {
-                    return false;
-                }
+            var supportsFeatureService = document.Project.Solution.Workspace.Services.GetService<IDocumentSupportsFeatureService>();
+            if (!supportsFeatureService.SupportsRefactorings(document))
+            {
+                return false;
+            }
 
-                // Finish any rename that had been started. We'll do this here before we enter the
-                // wait indicator for Extract Method
-                if (_renameService.ActiveSession != null)
-                {
-                    _renameService.ActiveSession.Commit();
-                }
+            // Finish any rename that had been started. We'll do this here before we enter the
+            // wait indicator for Extract Method
+            if (_renameService.ActiveSession != null)
+            {
+                _renameService.ActiveSession.Commit();
+            }
```

Comment:
this needs to be front loaded.  This was mentioned in the original PR but was ignored.